document/src/vespa/document/fieldvalue/fieldvalue.h

```diff
@@ -84,6 +84,13 @@ class FieldValue : public vespalib::Identifiable
     virtual int compare(const FieldValue& other) const;
 
     /**
+     * Same as normal compar, but this one expects the types to be equal
+     * @param other
+     * @return See compare
+     */
+    virtual int fastCompare(const FieldValue& other) const;
```

Comment:
I think it would be beneficial to give this function a name which more directly states its preconditions, e.g. `compareSameTypes`, `identicalTypeCompare` or something like that.